src/components/edit-event-form.js

```diff
@@ -274,12 +315,56 @@ class EditEventForm extends AbstractSmartComponent {
     this.rerender();
   }
 
+  getFormData() {
+    const form = this.getElement();
+    const eventDescription = form.querySelector(`.event__destination-description`).innerText;
+    const formData = new FormData(form);
+    const eventName = form.querySelector(`.event__type-output`).innerText.trim().split(` `)[0];
+    const isActivityEvent = ACTIVITY_EVENTS.some((it) => it === eventName);
+    const eventPhotos = [].map.call(form.querySelectorAll(`.event__photo`), (it) => it.src);
+    const eventOffers = this._event.offers;
+
+    const newPoint = {
+      id: this._event.id,
+      type: isActivityEvent ? `activity` : `transfer`,
+      name: eventName,
+      icon: `${eventName}.png`,
+      startDate: parseDateWithSlashes(formData.get(`event-start-time`)),
+      endDate: parseDateWithSlashes(formData.get(`event-end-time`)),
+      destination: formData.get(`event-destination`),
+      description: eventDescription,
+      price: formData.get(`event-price`),
+      photos: eventPhotos,
+      offers: eventOffers,
+      isFavorite: false
+    };
+
+    return newPoint;
+  }
+
+  _showDetails() {
+    this.getElement().querySelector(`.event__details`).classList.remove(HIDE_CLASS);
+  }
+
+  _setValidation() {
+    const startDateInput = this._element.querySelector(`input[name=event-start-time]`);
+
+    if (getDatesDiff(this._event.startDate, this._event.endDate) > 0) {
+      startDateInput.setCustomValidity(`The start time should be earlier than the end time`);
+    } else {
+      startDateInput.setCustomValidity(``);
+    }
+  }
+
+
   _subscribeOnEvents() {
     const element = this.getElement();
 
-    element.querySelector(`.event__favorite-checkbox`).addEventListener(`change`, () => {
-      this._event = Object.assign({}, this._event, {isFavorite: !this._event.isFavorite});
-    });
+    if (this._mode !== Mode.ADD) {
```

Comment:
```suggestion
    if (this._mode === Mode.EDIT) {
```
это не смертельно, но просто условия с равно читаются лучше исловий с не равно